compiler-core/src/format/tests/binary_operators.rs

```diff
@@ -0,0 +1,172 @@
+use crate::assert_format;
+
+// https://github.com/gleam-lang/gleam/issues/835
+#[test]
+pub fn long_binary_operation_sequence() {
+    assert_format!(
+        r#"pub fn main() {
+  int.to_string(color.red)
+  <> ", "
+  <> int.to_string(color.green)
+  <> ", "
+  <> int.to_string(color.blue)
+  <> ", "
+  <> float.to_string(color.alpha)
+}
+"#
+    );
+}
+
+#[test]
+pub fn long_comparison_chain() {
+    assert_format!(
+        r#"pub fn main() {
+  trying_a_comparison(this, is, a, function) > with_ints
+  && trying_other_comparisons < with_ints
+  || trying_other_comparisons <= with_ints
+  && trying_other_comparisons >= with_ints
+  || and_now_an_equality_check == with_a_function(foo, bar)
+  && trying_other_comparisons >. with_floats
+  || trying_other_comparisons <. with_floats(baz)
+  && trying_other_comparisons <=. with_floats
+  || trying_other_comparisons(foo, bar) >=. with_floats
+  && foo <> bar
+}
+"#
+    );
+}
+
+#[test]
+pub fn long_chain_mixing_operators() {
+    assert_format!(
+        r#"pub fn main() {
+  variable + variable - variable * variable / variable
+  == variable * variable / variable - variable + variable
+  || foo * bar > 11
+}
+"#
+    );
+
+    assert_format!(
+        r#"pub fn main() {
+  variable +. variable -. variable *. variable /. variable
+  == variable *. variable /. variable -. variable +. variable
+  || foo *. bar >=. 11
+}
+"#
+    );
+}
+
+// Thanks Hayleigh for pointing this out!
+#[test]
+fn case_branch_is_not_broken_if_can_fit_on_line() {
+    assert_format!(
+        r#"pub fn main() {
+  case remainder {
+    _ if remainder >=. 0.5 && x >=. 0.0 ->
+      float_sign(x) *. truncate_float(xabs +. 1.0) /. p
+    _ -> float_sign(x) *. xabs_truncated /. p
+  }
+}
+"#
+    );
+}
+
+// https://discord.com/channels/768594524158427167/1187508793945378847/1187508793945378847
+#[test]
+fn binary_operation_in_assignment_that_is_almost_80_chars() {
+    assert_format!(
+        r#"pub fn main() {
+  let is_vr_implicit =
+    dicom_read_context.transfer_syntax == transfer_syntax.ImplicitVrLittleEndian
+}
+"#
+    );
+}
+
+// https://github.com/gleam-lang/gleam/issues/2480
+#[test]
+fn labelled_field_with_binary_operators_are_not_broken_if_they_can_fit() {
+    assert_format!(
+        r#"pub fn main() {
+  Ok(Lesson(
+    name: names.name,
+    text: text,
+    code: code,
+    path: chapter_path <> "/",
+    previous: None,
+    next: None,
+  ))
+}
+"#
+    );
+
+    assert_format!(
+        r#"pub fn main() {
+  Ok(Lesson(
+    name: names.name,
+    text: text,
+    code: code,
+    path: chapter_path
+      <> "/"
```

Comment:
By any chance, would you mind explaining how the formatter knows to indent here but knows _not_ to indent on line 9 of this file?

I'm trying to follow along with this PR, but I'm new to this formatter code. Thanks!

Reply:
If you check the call_arg function in the formatter it add a layer of indentation for binary operators

Reply:
@jonmellman if you want to chat about pretty printing you can reach out in the Gleam discord server! I'd be more than happy to share what I've learned about the topic 💜